packages/react-router/docs/api/withRouter.md

```diff
@@ -27,3 +27,72 @@ class ShowTheLocation extends React.Component {
 // terminology) to the router.
 const ShowTheLocationWithRouter = withRouter(ShowTheLocation)
 ```
+
+**IMPORTANT NOTE:** The withRouter function manipulates React's ```shouldComponentUpdate``` lifecycle, and will collide with other programs using the method, **especially Redux**. If your code looks like the below example, your component will **not** update when the router changes.
```

Comment:
@pshrmn I've added the comment you mentioned and just added your example. That should work for most users who get stuck here.